src/azure-cli-core/azure/cli/core/commands/paramutil.py

```diff
@@ -0,0 +1,55 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.parameters import register_cli_argument, ignore_type
+from azure.cli.core.commands.validators import get_complex_argument_processor
+from azure.cli.core.commands._introspection import (extract_args_from_signature,
+                                                    _option_descriptions)
+from azure.cli.core.commands import _cli_extra_argument_registry
+
+
+class ParametersContext(object):
+    def __init__(self, command):
+        self._commmand = command
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def ignore(self, argument_name):
+        register_cli_argument(self._commmand, argument_name, ignore_type)
+
+    def argument(self, argument_name, arg_type=None, **kwargs):
+        register_cli_argument(self._commmand, argument_name, arg_type=arg_type, **kwargs)
+
+    def expand(self, argument_name, model_type, group_name=None, patches=None):
```

Comment:
The "Account SAS Arguments" group and the "Service SAS Arguments" groups aren't particularly helpful distinctions because they basically encompass all arguments. It think it would be more helpful to break out optional groups of arguments like Content Settings or the parameters which look like they clearly only apply to tables (row-key/partition-key start/end).